tests/unit/test_workspace.py

```diff
@@ -50,29 +51,46 @@ def test_compound_indexer(tmpdir, pea_id):
 
 def test_compound_indexer_rw(tmpdir):
     all_vecs = np.random.random([6, 5])
+    executor_yml = f'''!CompoundExecutor
+components:
+  - !BinaryPbIndexer
+    with:
+      index_filename: metaproto
+    metas:
+      name: test_meta
+  - !NumpyIndexer
+    with:
+      metric: euclidean
+      index_filename: npidx
+    metas:
+      name: test_numpy
+metas:
+  name: real-compound
+  workspace: {tmpdir}
+    '''
     for j in range(3):
-        with BaseExecutor.load_config('yaml/test-compound-indexer2.yml', separated_workspace=True, pea_id=j) as a:
-            assert a[0] == a['test_meta']
-            assert not a[0].is_updated
-            assert not a.is_updated
-            a[0].add([j, j * 2, j * 3], [bytes(j), bytes(j * 2), bytes(j * 3)])
-            a[0].add([j, j * 2, j * 3], [bytes(j), bytes(j * 2), bytes(j * 3)])
-            assert a[0].is_updated
-            assert a.is_updated
-            assert not a[1].is_updated
-            a[1].add(np.array([j * 2, j * 2 + 1]), all_vecs[(j * 2, j * 2 + 1), :])
-            assert a[1].is_updated
-            a.save()
+        with BaseExecutor.load_config(executor_yml, separated_workspace=True, pea_id=j) as indexer:
```

Comment:
@bwanglzu @JoanFM How do you like to put the yml string directly into the test. Advantages: much better readable without file jumping and you can adapt the workspace without first writing to env variable and then read again. This is only a good approach for yaml with <20 lines, but here I really prefer it much.

Reply:
make sense to me

Reply:
is this even possible to have a basic `yml` and pass extra parameters? even without the metas?

Reply:
For me having this is kind of perfect, I kind of got used to have `yml` files independently but it is okey for me to have it embedded in the code, but when it comes to `flow` yamls with several `pods` I do believe is better to have it as independent file

Reply:
I like this a lot since it is much more explicit what is gone tested.
The only down site I see is that we don't have syntax highlighting and auto format anymore.
But since the files are very small, this won't hurt to0 much.